src/client/src/Pages/CreateJob.tsx

```diff
@@ -1,7 +1,179 @@
+    label: "Biology",
+    value: "biology"
+  },
+  {
+    label: "Biochemistry",
+    value: "biochemistry"
+  },
+  {
+    label: "Bioengineering",
+    value: "bioengineering"
+  },
+  {
+    label: "Chemistry",
+    value: "chemistry"
+  },
+  {
+    label: "Chemical Engineering",
+    value: "chemical_engineering"
+  },
+  {
+    label: "Computer Science",
+    value: "computer_science"
+  },
+  {
+    label: "Computer Engineering",
+    value: "computer_engineering"
+  },
+  {
+    label: "Data Science",
+    value: "data_science"
+  },
+  {
+    label: "Environmental Engineering",
+    value: "environmental_engineering"
+  },
+  {
+      label: "Electrical Engineering",
+      value: "electrical_engineering"
+  },
+  {
+    label: "Mathematics",
+    value: "mathematics"
+  },
+  { 
+      label: "Materials Science and Engineering",
+      value: "materials_sci_engineering"
+  },
+  {
+      label: "Mechanical Engineering",
+      value: "mechanical_engineering"
+  },
+  {
+    label: "Physics",
+    value: "physics"
+  },
+  {
+    label: "Statistics",
+    value: "statistics"
+  }
+];
+
+// Regex for numeric checker
+const numericRegex = /^\d+$/; 
+
+// Input Validation
+const SignupSchema = Yup.object().shape({
+  hoursPerWeek: Yup.string()
+  .matches(numericRegex, 'All numerics required!')
+  .required('Required!')
+}) 
+
+const CreateJob: React.FC = () => {
+  const handleSubmit = (values: FormValues): void => {
+    alert(JSON.stringify(values));
+  };
+
+  return (
+    <div className="App">
+      <h1>Create A Job Post</h1>
+      <Formik
+        initialValues={initialValues}
+        onSubmit={handleSubmit}
+        validationSchema={SignupSchema}
+      >
+        {({ dirty, isValid }) => {
+          return (
+            <Form>
+              <FormikField name="title" label="Title" required />
+              <div id="checkbox-group">Target Years</div>
+            <div role="group" aria-labelledby="checkbox-group">
+              <label>
+                <Field type="checkbox" name="targetYears" value="freshman" />
+                Freshman
+              </label>
+              <label>
+                <Field type="checkbox" name="targetYears" value="sophomore" />
+                Sophomore
+              </label>
+              <label>
+                <Field type="checkbox" name="targetYears" value="junior" />
+                Junior
+              </label>
+              <label>
+                <Field type="checkbox" name="targetYears" value="senior" />
+                Senior
+              </label>
+            </div>
+            <FormikField name="hoursPerWeek" label="Hours Per Week" required />
+              <FormikSelect
+                name="department"
+                items={DepartmentItems}
+                label="Department"
+                required
+              />
+              <FormikField name ="type" label="Type" required/>
+              <FormikField name ="startDate" label="Start Date" required/>
```

Comment:
Here you can use the "type" prop like this type="date", so that this text field only accepts date inputs Same for expirationDate and endDate when you add it